web/src/message_list_data.ts

```diff
@@ -160,68 +172,70 @@ export class MessageListData {
         return last_msg.id === this._selected_id;
     }
 
-    nth_most_recent_id(n) {
+    nth_most_recent_id(n: number): number {
         const i = this._items.length - n;
         if (i < 0) {
             return -1;
         }
         return this._items[i].id;
     }
 
-    clear() {
+    clear(): void {
         this._all_items = [];
         this._items = [];
         this._hash.clear();
     }
 
-    get(id) {
-        id = Number.parseFloat(id);
+    get(id_string: string): Message | undefined {
+        const id = Number.parseFloat(id_string);
```

Comment:
We definitely call this with a `number` elsewhere, e.g. in `condense_and_collapse` and `MessageList#selected_message` and `message_hover`. Ideally, it should always be a `number` and we can delete the overdefensive `parseFloat`. But that will be easier to audit once everything is TypeScript, so for now, I guess `number | string` is safest.